src/CMakeLists.txt

```diff
@@ -326,13 +326,15 @@ if(NOT TARGET RapidJSON::RapidJSON)
 endif()
 
 option(WITH_FMT_HEADER_ONLY "use header-only version of fmt library" OFF)
-set(WITH_FMT_VERSION "8.1.1" CACHE
-  STRING "build with fmt version")
-find_package(fmt ${WITH_FMT_VERSION} QUIET)
+find_package(fmt 8.1.1...<10.0.0)
 if(fmt_FOUND)
-  include_directories(SYSTEM "${fmt_INCLUDE_DIR}")
+  message(STATUS "Building with system fmt.")
+  if (WITH_FMT_HEADER_ONLY)
+    message(STATUS "Using fmt header-only.")
+    add_library(fmt::fmt ALIAS fmt::fmt-header-only)
```

Comment:
i am fine with removing `Findfmt.cmake`, but i don't quite understand 

> It also removes the `Findfmt.cmake` module, as that does not check
> the installed version. Instead, we use the cmake config file installed by
> the system package of fmt and does support version checking.
> 

because `Findfmt.cmake` does parse the macro of `FMT_VERSION`, and expose the parse result.

Reply:
It exposes it, but I think it's better if we can just pass the version into find_package and have find_package check it. That we were passing the version to find_package makes me think that was the expected behavior.